### PR TITLE
Force fail on all unimplemented methods.

### PR DESCRIPTION
**Pull request by vanbasten23**
This pr makes it easy to debug dynamic shape related issues.

### CODE REVIEW

File: torch_xla/csrc/tensor.cpp [hunk 1]
```diff
@@ -603,14 +603,42 @@ bool XLASymNodeImpl::is_float() {
   return false;
 }
 
-bool XLASymNodeImpl::bool_() {
-  auto dn = torch_xla::DimCast(node());
-  return dn->getDynamicValue() != 0;
+c10::SymNode XLASymNodeImpl::add(const c10::SymNode& other) {
+  auto pother = dynamic_cast<XLASymNodeImpl*>(other.get());
```
Comment (JackCaoG) [inline torch_xla/csrc/tensor.cpp:607]: nit `p_other` nad `n_add`

**Review by JackCaoG — COMMENTED** [review]
hmm, does upstream has a defualt implementation? I would expect upstream just failed if it is unimplemented.

**Comment by vanbasten23** [comment]
> hmm, does upstream has a defualt implementation? I would expect upstream just failed if it is unimplemented.

I agree, I'd expect it just fails if we haven't implemented the method for XLASymNodeImpl. 

I added it here because Ed in his [comment](https://github.com/pytorch/xla/pull/4289#issuecomment-1348387865) mentioned `The easiest is to just edit each virtual method in the XLA SymNode subclass to print stuff. We don't have a builtin printer in core but it is also trivial to write for us as our SymNode is written in Python. I suppose we could also write an adapter lol`. I'm reluctant to edit each virtual methods every time we debug a dynamic shape related issue. And we may need to implement them eventually.

File: torch_xla/csrc/tensor.cpp (hunk 1 above)
Comment (vanbasten23) [reply torch_xla/csrc/tensor.cpp]: done

File: torch_xla/csrc/tensor.cpp [hunk 2]
```diff
@@ -603,43 +603,139 @@ bool XLASymNodeImpl::is_float() {
   return false;
 }
 
-bool XLASymNodeImpl::bool_() {
-  auto dn = torch_xla::DimCast(node());
-  return dn->getDynamicValue() != 0;
+c10::SymNode XLASymNodeImpl::add(const c10::SymNode& other) {
+  auto p_other = dynamic_cast<XLASymNodeImpl*>(other.get());
+  auto n_add = torch::lazy::MakeNode<SizeAdd>(node(), p_other->node());
+  return c10::make_intrusive<XLASymNodeImpl>(n_add);
```
Comment (JackCaoG) [inline torch_xla/csrc/tensor.cpp:609]: can we add test for this method?

Comment (vanbasten23) [reply torch_xla/csrc/tensor.cpp:609]: Another [pr](https://github.com/pytorch/xla/pull/4258) is intended to add tests for this method and the current pr just wants to force fail the unimplemented method. Do you mind if I add the test in the other pr?

Comment (JackCaoG) [reply torch_xla/csrc/tensor.cpp:609]: oh ok, you did not modify this implemention, it is just a git diff thing.

**Review by JackCaoG — APPROVED** [review]
Thanks!

**Pull request merged**